main.go

```diff
@@ -232,12 +239,60 @@ func main1() int {
 	return 0
 }
 
+func checkGoVersion() bool {
+	out, err := exec.Command("go", "version").CombinedOutput()
+	if err != nil {
+		fmt.Printf(`Can't get go version: %v
+
+Possible reasons:
+1. Golang not installed
+2. Environment variables $PATH or $GOROOT set incorrectly
+3. Unsupported golang version installed (supported %s)
+`, err, supportedGoVersion)
+		return false
+	}
+
+	rawVersion := string(bytes.TrimPrefix(bytes.TrimSpace(out), []byte("go version ")))
+	if strings.Contains(rawVersion, "weekly") || strings.Contains(rawVersion, "devel") {
+		fmt.Printf(`You use the unstable "%s" golang version.
+Garble has not been tested on this version, recommend using supported golang version %s`, rawVersion, supportedGoVersion)
+		return true
+	}
+
+	versionStr := regexp.MustCompile(`go([1-9][0-9]*)(\.(0|[1-9][0-9]*))?`).FindString(rawVersion)
+	if versionStr == "" {
+		fmt.Printf(`Can't recognize golang version: %s
+
+Possible reasons:
+1. Incorrectly installed golang
+2. Unsupported golang version installed (supported %s)
+`, versionStr, supportedGoVersion)
+		return false
+	}
+
+	// Error while parsing version is impossible because earlier the string was checked through regexp
+	version, _ := strconv.ParseFloat(strings.TrimPrefix(versionStr, "go"), 64)
+	if version < minGoVersion {
+		fmt.Printf("Outdated golang version %.2f is used, please upgrade golang to %s", version, supportedGoVersion)
+		return false
+	}
+	if version >= maxGoVersion {
+		fmt.Printf("Too new golang version %.2f is used, please downgrade golang to %s or try upgrade garble", version, supportedGoVersion)
```

Comment:
Current go version %.2f is to new, please downgrade go to %s or try to upgrade garble.